models.cpp

```diff
@@ -596,41 +610,50 @@ std::valarray<double> Models::ThreeStepAlternative::right_hand_side(const std::v
 
 // class to hold integration hyperparameters
 Models::explEulerParameters::explEulerParameters(const double startTimeValue,
-                      const double endTimeValue,
+                      const std::vector<double> outputTimesValue,
                       const std::valarray<double> initialConditionValues)
 {
   startTime = startTimeValue;
-  endTime = endTimeValue;
+  outputTimes = outputTimesValue;
   initialCondition = initialConditionValues;
 }
 
 
 
 // ODE solver
-std::valarray<double> Models::integrate_ode_explicit_euler(const Models::explEulerParameters solverParameters,
-                                                   const Models::ModelsBase& model,
-                                                   const Models::ParametersBase& modelParameters)
+std::vector<std::valarray<double>> Models::integrate_ode_explicit_euler(const Models::explEulerParameters solverParameters,
+                                                           const Models::ModelsBase& model,
+                                                           const Models::ParametersBase& modelParameters,
+                                                           const double time_step)
 {
+  const std::vector<double> evalTimes = solverParameters.outputTimes;
```

Comment:
You are, in essence, creating a duplicate of the variable you already have. Why not work with the original variable instead?